Make the go program easier to use as a standalone program: make it a go module, add command line flags and support for ddnskey

Golit opened the pull request:
This pull request will make the go program easier to use as a standalone program.

* Make the go source a go module since it is easier to use if used independently from the Dockerfile.
* The pull request removes the necessity of having a config file. All configurations can be passed with command line arguments.
* Some duplicate code is removed and refactored. Especially all copies of `RequestDataExtractor` where moved to a single file.
* Middleware architecture is used to gather user input variables which should make it more flexible.
* Changed go compiler to 1.14.

### `nsupdate` improvements
* Commands to `nsupdate` are now piped to the program.
* A `ddnskey` can be passed from the http request to the `nsupdate` program.
* All commands send to the `nsupdate` program can be logged.
* User input values are escaped before passing them to the `nsupdate` program.

akune committed:
feat: allow updating ip4 and ip6 addresses in one request

Golit committed:
Add go module files

* This enables the program to be compiled outside the go source
  directory and makes the project easily usable independently

Golit committed:
Replace encoded base64 value to clear text

Golit committed:
Add support for command line arguments

* Add port configuration value

Golit committed:
Refactor nsupdate function into a new file

* Update nsupdate code to use pipe instead of creating a new file
* Add escaping for user input variables
* Add interface for DNS update client implementations

Golit committed:
Change builder to golang:1.14

Golit committed:
Refactor requestDataExtractor and add ddnskey

Golit committed:
Remove dublicate code

Golit committed:
Merge commit '60ba35e' into develop

Golit committed:
Fix json ouput message formating

Golit committed:
Fix nsupdate call if zone is empty

Golit committed:
Replace error message variables

* It is more usefull to have the user input value in the error message

dprandzioch commented:
Hey @Golit thanks for that huge contribution. I'll have a look at it as soon as I find some time for that!

Golit committed:
Change sharedSecred checking method

The rest-api does not need to check the shared secret because bind itself
can check it. This change also allows to have different shared secrets for
different zones.

See #55

Golit committed:
Add support for multiple zones

* Fix #7 #18 by allowing updates by fqdn

Golit committed:
Add support to update any record type

* Any record type can be updated by passing query parameter 'type' and
  'value' or 'addr' as a fallback
* Some code cleanup

Closes #49
Closes #61

Golit committed:
Add delete endpoint
